src/Units/UnitState.java

```diff
@@ -50,17 +50,34 @@ public UnitState(UnitState other)
   {
     CO = other.CO;
     model = other.model;
-    ammo = other.ammo;
-    fuel = other.fuel;
-    materials = other.materials;
     isTurnOver = other.isTurnOver;
-    health = other.health;
+    setResourceState(other);
     captureProgress = other.captureProgress;
     captureTarget = other.captureTarget;
 
     heldUnits = new CargoList(model);
     heldUnits.addAll(other.heldUnits);
   }
+  public void setResourceState(UnitState other)
```

Comment:
This feels like an overly general name. Maybe `copyUnitState` would be clearer?

Reply:
Done in PR2